### PR TITLE
Audiencerun bid adapter

### PR DESCRIPTION
**Pull request by audiencerun**
## Type of change
- [x] New bidder adapter   

## Description of change
- New adapter for Audiencerun.com Ad Network
- test parameters for validating bids
```
{
  bidder: 'audiencerun',
  params: { 
              zoneId: 'xtov2mgij0'
          }
}
```

- link to a PR on the docs repo https://github.com/prebid/prebid.github.io/pull/1756

### CODE REVIEW

File: modules/audiencerunBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+   * Make a server request from the list of BidRequests.
+   *
+   * @param {BidRequest[]} bidRequests A non-empty list of bid requests which should be sent to the Server.
+   * @return ServerRequest Info describing the request to the server.
+   */
+  buildRequests: function(bidRequests, bidderRequest) {
+    const bids = bidRequests.map(bid => {
+      return {
+        zoneId: utils.getValue(bid.params, 'zoneId'),
+        sizes: bid.sizes.map(size => ({
```
Comment (msm0504) [inline modules/audiencerunBidAdapter.js]: `sizes` is no longer supported in Prebid 3. `mediaTypes.banner.sizes` should be used instead.

File: test/spec/modules/audiencerunBidAdapter_spec.js [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+  });
+
+  describe('isBidRequestValid', function() {
+    let bid = {
+      'bidder': 'audiencerun',
+      'params': {
+        'zoneId': '12345abcde'
+      },
+      'adUnitCode': 'adunit-code',
+      'sizes': [[300, 250], [300, 600]],
```
Comment (msm0504) [inline test/spec/modules/audiencerunBidAdapter_spec.js]: `sizes` is no longer supported in Prebid 3. `mediaTypes.banner.sizes` should be used instead.

File: test/spec/modules/audiencerunBidAdapter_spec.js [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+    let bidRequests = [
+      {
+        'bidder': 'audiencerun',
+        'bidId': '51ef8751f9aead',
+        'params': {
+          'zoneId': '12345abcde'
+        },
+        'adUnitCode': 'div-gpt-ad-1460505748561-0',
+        'transactionId': 'd7b773de-ceaa-484d-89ca-d9f51b8d61ec',
+        'sizes': [[320, 50], [300, 250], [300, 600]],
```
Comment (msm0504) [inline test/spec/modules/audiencerunBidAdapter_spec.js]: `sizes` is no longer supported in Prebid 3. `mediaTypes.banner.sizes` should be used instead.

File: modules/audiencerunBidAdapter.js [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+    const payload = {
+      libVersion: this.version,
+      referer: bidderRequest.refererInfo ? bidderRequest.refererInfo.referer || null : null,
+      currencyCode: config.getConfig('currency.adServerCurrency'),
+      timeout: config.getConfig('bidderTimeout'),
+      bids
+    };
+
+    if (bidderRequest && bidderRequest.gdprConsent) {
```
Comment (msm0504) [inline modules/audiencerunBidAdapter.js]: Will you add support for CCPA or will you only support GDPR?

File: modules/audiencerunBidAdapter.js [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+  /**
+   * Register the user sync pixels which should be dropped after the auction.
+   *
+   * @param {SyncOptions} syncOptions Which user syncs are allowed?
+   * @param {ServerResponse[]} serverResponses List of server's responses.
+   * @return {UserSync[]} The user syncs which should be dropped.
+   */
+  getUserSyncs: function(syncOptions, serverResponses) {
+    if (syncOptions.pixelEnabled) {
```
Comment (msm0504) [inline modules/audiencerunBidAdapter.js]: `pixelEnabled ` is no longer supported in Prebid 3

File: modules/audiencerunBidAdapter.js [hunk 6]
```diff
@@ -0,0 +1,143 @@
+import * as utils from '../src/utils';
+import { config } from '../src/config';
+import { registerBidder } from '../src/adapters/bidderFactory';
+import { BANNER } from '../src/mediaTypes';
+
+const BIDDER_CODE = 'audiencerun';
+const ENDPOINT_URL = 'https://d.audiencerun.com/prebid';
```
Comment (msm0504) [inline modules/audiencerunBidAdapter.js]: Responses from this endpoint are not compressed, but are required to be. 

Adapter Requirements: http://prebid.org/dev-docs/bidder-adaptor.html#bidder-adaptor-Required-Adapter-Conventions

Comment (audiencerun) [reply modules/audiencerunBidAdapter.js]: we are already return content-encoding "gzip"

File: modules/audiencerunBidAdapter.js (hunk 4 above)
Comment (audiencerun) [reply modules/audiencerunBidAdapter.js]: not immediately

File: modules/audiencerunBidAdapter.js [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+
+  /**
+   * Make a server request from the list of BidRequests.
+   *
+   * @param {BidRequest[]} bidRequests A non-empty list of bid requests which should be sent to the Server.
+   * @return ServerRequest Info describing the request to the server.
+   */
+  buildRequests: function(bidRequests, bidderRequest) {
+    const bids = bidRequests.map(bid => {
+      const sizes = utils.deepAccess(bid, 'mediaTypes.banner.sizes');
```
Comment (msm0504) [inline modules/audiencerunBidAdapter.js]: Is it possible for a bid to have null or undefined as its banner sizes? If so, add `[]` as a 3rd parameter (default value). That will prevent errors when calling `map` on line 40

**Comment by audiencerun** [comment]
Thank's for you review msm0504
Do you need anything else to validate?

**Comment by msm0504** [comment]
@audiencerun No, I don't need anything else. This will be in the next Prebid release

**Pull request merged**